e2e/cypress/integration/menus/status_dropdown_spec.js

```diff
@@ -36,6 +48,106 @@ describe('Status dropdown menu', () => {
         cy.get('.MenuWrapper.status-dropdown-menu .Menu__content.dropdown-menu').should('be.visible');
     });
 
+    it('MM-T671 Prompting to set status to online from Away', () => {
+        cy.apiLogin(user1);
+        cy.visit(testChannelUrl);
+
+        // # Use the status drop-down on your profile pic to go Away
+        cy.get('img.Avatar').click();
+        cy.findByText('Away').click();
+
+        // * Your status stays offline in your view
+        cy.get('.away--icon').should('be.visible');
+
+        // # Log out
+        cy.apiLogout();
+
+        // # Log back in
+        cy.apiLogin(user1);
+        cy.visit(testChannelUrl);
+
+        // # On modal that asks if you want to be set Online, select Yes.
+        cy.get('.modal-content').within(() => {
+            cy.findByText('Your Status is Set to "Away"').should('be.visible');
+            cy.findByText('Yes, set my status to "Online"').click();
+        });
+
+        // * Your status is online in your view
+        cy.get('.online--icon').should('be.visible');
+
+        // # login in as user2
+        cy.apiLogin(user2);
+        cy.visit(testChannelUrl);
+        openDM(user1.username);
+
+        // * Your status is online in other users' views.
+        cy.apiGetUserStatus(`${user1.id}`).then((result) => {
+            cy.wrap(result.status.status).should('be.equal', 'online');
+        });
```

Comment:
Unfortunately, I couldn't get this working without forcing `user1` to update its status to online externally
```javascript
cy.apiLogin(user2);
cy.visit(testChannelUrl);
cy.makeClient({user: user1}).then((client) => {
    cy.wrap(client.updateStatus({
        user_id: user1.id,
        status: 'online',
        manual: true,
    })).then((response) => {
        expect(response.status).to.equal('online');
        openDM(user1.username);

        // * Your status is online in other users' views.
        cy.apiGetUserStatus(user1.id).its('status.status').should('equal', 'online');
    });
});
```
Note: There's a fix for `cy.makeClient` if you'd like to try above - https://github.com/mattermost/mattermost-webapp/pull/7227/files